scripts/specify_operator_update.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+"""
+Script to add replaces and skips sections to Pelorus Operator
+ClusterServiceVersion, located in
+
+../pelorus-operator/bundle/manifests/pelorus-operator.clusterserviceversion.yaml
+
+More info: https://olm.operatorframework.io/docs/concepts/olm-architecture/operator-catalog/creating-an-update-graph/
+"""
+
+import argparse
+import sys
+from pathlib import Path
+from typing import List
+
+YAML_TAB = "  "
+
+
+def path_type(file_path: str) -> Path:
+    _file_path = Path(file_path).resolve()
+    if _file_path.exists():
+        return _file_path
+    raise argparse.ArgumentTypeError(f"Folder {_file_path} does not exist.")
+
+
+def exit_error(message: str) -> None:
+    print(f"ERROR: {message}", file=sys.stderr)
+    raise SystemExit(1)
+
+
+def main(
+    new_version: str, cluster_service_version_file: Path, versions: List[str]
+) -> None:
+    parsed_versions = [
+        version
```

Comment:
Should we use here:

https://python-semver.readthedocs.io/en/stable/usage.html#checking-for-a-valid-semver-version

+ check that `prerelease` and `build` are None:

https://python-semver.readthedocs.io/en/stable/usage.html#accessing-parts-of-a-version-through-names

Then we can use max() to find out the newest version...

Reply:
This way we can ignore `head -1` and use your script to parse output versions ?

Reply:
the script is called at the end of the create script, it would had to be another one in my opinion. But calling two python scripts in the middle of a bash one, does not seem good, it is better to write everything in Python (and it would be much easier to filter the tags with Python, like the example you gave, I had to use regex with bash)

Reply:
@mateusoliveira43 now it's really working great. One corner case which I found and don't know if it's worth updating at this point:

If we want to re-spin (recreate) already existing tag, for example during our review process we found that 0.0.2 needs fixing, then we would use:

```shell
./scripts/create_pelorus_operator -v 0.0.2
```

pelorus-operator files are properly generated (as expected), but then at the end we get:
```shell
ERROR: version 0.0.2 already exists in Pelorus Operator tags.
```

So, our spec looks like:
```diff
@@ -461,6 +461,3 @@ spec:
     name: Red Hat
     url: https://redhat.com
   version: 0.0.2
-  replaces: pelorus-operator.v0.0.1
-  skips:
-    - pelorus-operator.v0.0.1
```

I think we should ideally error *before* we generate any files or allow to run the script and generate `skips/replaces` giving at the end `WARNING`, that this tag is already in the repo. This is valid case as I wrote, when someone would like to spin off same version for any reason.

Reply:
Ok I am happy with this approach.